src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java

```diff
@@ -328,11 +338,20 @@ public void testCreateCheckWithPartialPackageNameByBruteForce() throws Exception
     @SuppressWarnings("unchecked")
     public void testGenerateThirdPartyNameToFullModuleNameWithException() throws Exception {
         final URLClassLoader classLoader = mock(URLClassLoader.class);
-        when(classLoader.getURLs()).thenThrow(IOException.class);
-        final Method method = factory.getClass().getDeclaredMethod(
+        final Set<String> packages = Collections.singleton(BASE_PACKAGE);
+
+        // Set up an exception in the ModuleReflectionUtil.getCheckstyleModules()
+        mockStatic(ModuleReflectionUtil.class);
+        doThrow(new IOException("mock exception")).when(ModuleReflectionUtil.class);
+        ModuleReflectionUtil.getCheckstyleModules(packages, classLoader);
+
+        // Invoke the method via reflection
```

Comment:
Please do me favor and place javadoc comment over the method to briefly explain why reflection is the only way to test it.  We try to keep it as rule now, and this method was created before to agreed on it. We try to make it clear that usage of reflection is allowed but not easy way for engineer.

Reply:
done.